src/PhpSpec/CodeGenerator/Generator/MethodGenerator.php

```diff
@@ -65,18 +78,21 @@ public function supports(Resource $resource, string $generation, array $data) :
 
     /**
      * @param Resource $resource
-     * @param array             $data
+     * @param array    $data
      */
     public function generate(Resource $resource, array $data = array())
     {
-        $filepath  = $resource->getSrcFilename();
-        $name      = $data['name'];
-        $arguments = $data['arguments'];
-
-        $argString = \count($arguments)
-            ? '$argument'.implode(', $argument', range(1, \count($arguments)))
-            : ''
-        ;
+        $filepath = $resource->getSrcFilename();
+        $name = $data['name'];
+
+        if (isset($data['reflection_parameters'])) {
```

Comment:
I'm not happy with this conditional - calling the method with 'reflection arguments' instead, when we want type hints, is not an obvious API.

Perhaps we should have an `arguments` array in all cases, and a separate optional `argument_types`? This would require the caller doing the reflection before calling the generator

It makes sense to me that the caller 'knows' the types come from analysing a method, but the generator is just using a list it's passed - will clean up the generator code / make tests less reliant on example classes

Reply:
Or possibly we should have an `Argument` value object - might be overkill

Reply:
I'm not happy with it either - I mentioned in a comment further up that it needs a refactor as part of bringing in type hints for all method generation. I have tried to keep all of type hinted method signature generator out of this class and in `PhpSpec\CodeGenerator\Generator\Argument\StringBuilder`, so it can be re-used easily after refactoring.

It's not as straightforward as just adding a new `argument_types` key, as we would need to map nullable and default values along with the types - also I've tried to avoid the `$argument1, $argument2 ...` style of naming with this generation as we know the interface method argument names.

Reply:
It sounds then like `arguments` should contain value objects that capture this complexity (and can have `__toString` to support BC)